default/python/AI/ResourcesAI.py

```diff
@@ -551,14 +553,48 @@ def set_planet_industry_and_research_foci(focus_manager, priority_ratio):
                     focus_manager.bake_future_focus(pid, RESEARCH, False)
                 continue
             if adj_round == 3:  # take research at planets where can do reasonable balance
-                if has_force or foAI.foAIstate.character.may_dither_focus_to_gain_research() or (target_rp >= priority_ratio * cumulative_pp):
+                # if this planet in range where temporary Research focus ("research dithering") can get some additional
+                # RP at a good PP cost, and still need some RP, then consider doing it.
```

Comment:
I would really like an introductory comment that explains what "research dithering" actually is supposed to do: Abuse the fact that meters fall only by the same rate that they grow. So the fundamental idea is to grow to the research target to maximum while the PP meter is still above its target value at research focus target. 

Also, some explicit math wouldn't hurt. Something like
```
Assuming we switch to a research focus for n turns with a focus penalty of p and assuming integer values of Research(R) and Industry(I) only:

R = {sum(i), i=1..n}  = n(n+1)/2
I = {-sum(i+p), i=1..n} = -n(n+1+2p)/2

Growing up again until industry meter maximum, we need (n+p) turns.
R = {sum(i), i = 0..n-p-1} = (n-p-1)(n-p)/2
I = {-sum(i), i = 1..n+p-1} = (n+p-1)(n+p)/2

Adding both, we have
R = n^2 - np + p^2 / 2 + p/2
I = - (n^2 + 2np +p^2/2 -p/2)

With current p = 1 this yields
R = 1 + n(n-1)
I = - n(n+2)
or a cost ratio of -I/R = n(n+2) / (1+n(n-1))

For example, a cycle consisting of switching 5 turns to research yields (n=5)
R = 21
I = -35
or a cost of ~1.67 PP per RP
```